Drop end of life operating systems from data

ekohl opened the pull request:
This was checked against Puppet's supported agent configs[1].

Before I make this unreadable by also dropping the fixtures, I'd appreciate a review if I'm not dropping too much or too little.

[1]: https://www.puppet.com/docs/pe/2019.8/supported_operating_systems.html

ekohl commented on lib/beaker-hostgenerator/data.rb:
```diff
@@ -329,48 +251,6 @@ def osinfo
-            'template' => 'cisco-nxos_hw-9k-x86_64',
-          },
-        },
         'cisco_n9k-VM' => {
```
How much of cisco is still supported by Puppet?

bastelfreak replied:
Is there even any cisco still supported? I don't think so

ekohl replied:
Me neither, but perhaps someone can weigh in on this.

pmcmaw replied:
We no longer support cisco 👍

ekohl commented on lib/beaker-hostgenerator/data.rb:
```diff
@@ -680,19 +484,6 @@ def osinfo
-            'template' => 'huaweios-6-powerpc',
-          },
-        },
         'panos61-64' => {
```
Is this still supported?

ekohl replied:
@shaigy @pmcmaw can you also weigh in on this?

shaigy commented:
In PE, we still test agent upgrades on centos6 and osx10.15

pmcmaw commented:
In relation to supported module support, this PR looks in line with what we support.

mhashizume commented on lib/beaker-hostgenerator/data.rb:
```diff
@@ -829,62 +588,6 @@ def osinfo
-            'template' => 'osx-1014-x86_64',
-          },
-        },
-        'osx1015-64' => {
-          :general => {
-            'platform' => 'osx-10.15-x86_64',
-          },
-          :vmpooler => {
-            'template' => 'osx-1015-x86_64',
-          },
-        },
```
While [we've announced end of support for Catalina](https://groups.google.com/g/puppet-announce/c/Un-JAtMEvAc), we haven't quite removed it from our pipelines. Would you mind holding off on this for a little while longer?

ekohl replied:
So is that just 10.15 or all versions?

mhashizume replied:
Sorry, to be clear my preference would be to leave macOS 10.15 and up (10.15 Catalina, 11 Big Sur, 12 Monterey) as-is.

joshcooper replied:
Removing 10.15 is blocked on https://tickets.puppetlabs.com/browse/PA-5024. All of the prior versions can be deleted.

ekohl commented:
> In PE, we still test agent upgrades on centos6 and osx10.15

Is that only centos6 with x86_64 or also i386?

shaigy commented:
> > In PE, we still test agent upgrades on centos6 and osx10.15
> 
> Is that only centos6 with x86_64 or also i386?

Both, x86_64 and i386

GSPatton commented on lib/beaker-hostgenerator/data.rb:
```diff
@@ -520,82 +400,6 @@ def osinfo
-            'template' => 'debian-8-x86_64',
-          },
-        },
         'debian9-32' => {
```
I'm not sure if we still support debian 9? @pmcmaw @chelnak

pmcmaw replied:
Yeah we no longer do, however some other teams may still, as for removal of OSs in the PR as a whole, I think it aligns well with what we currently support.

ekohl replied:
I removed Debian 8 because it's gone from the Debian repos. I wasn't sure about Debian 9, so decided to be a bit more conservative. https://wiki.debian.org/LTS suggests the LTS ended 2022-06-30.

joshcooper replied:
We dropped agent support for Debian 9 in https://tickets.puppetlabs.com/browse/PA-4570. Feel free to remove it.

joshcooper commented on lib/beaker-hostgenerator/data.rb:
```diff
@@ -717,22 +508,6 @@ def osinfo
-          :vmpooler => {
-            'template' => 'opensuse-11-x86_64',
-          },
-        },
```
I think this is fine to remove, because I don't think we test against opensuse and we were only testing against SLES 11 in Puppet 6.x which is now EOL. Here is the corresponding removal ticket https://tickets.puppetlabs.com/browse/PA-5104

ekohl commented:
Thanks for the feedback. I think this addresses the concerns. It's still a draft since I haven't started with fixing the tests yet. That makes the PR very hard to review, so I'd prefer to first get an ACK on what to remove.
